src/aes.rs

```diff
@@ -14,59 +14,35 @@
 // You should have received a copy of the GNU Affero General Public License
 // along with this program.  If not, see <https://www.gnu.org/licenses/>.
 
+use aes::Aes256;
+use block_modes::block_padding::Pkcs7;
+use block_modes::{BlockMode, Cbc};
 use primitives::H256;
-use rcrypto::aes::KeySize::KeySize256;
-use rcrypto::aes::{cbc_decryptor, cbc_encryptor};
 use rcrypto::aessafe::AesSafe128Encryptor;
-use rcrypto::blockmodes::{CtrMode, PkcsPadding};
-use rcrypto::buffer::{BufferResult, ReadBuffer, RefReadBuffer, RefWriteBuffer, WriteBuffer};
+use rcrypto::blockmodes::CtrMode;
+use rcrypto::buffer::{RefReadBuffer, RefWriteBuffer};
 pub use rcrypto::symmetriccipher::SymmetricCipherError;
 use rcrypto::symmetriccipher::{Decryptor, Encryptor};
 
 use super::error::SymmError;
 
-fn is_underflow(result: BufferResult) -> bool {
-    match result {
-        BufferResult::BufferUnderflow => true,
-        BufferResult::BufferOverflow => false,
-    }
-}
+
+type Aes256Cbc = Cbc<Aes256, Pkcs7>;
 
 // AES-256/CBC/Pkcs encryption.
 pub fn encrypt(data: &[u8], key: &H256, iv: &u128) -> Result<Vec<u8>, SymmetricCipherError> {
-    let mut encryptor = cbc_encryptor(KeySize256, key, &iv.to_be_bytes(), PkcsPadding);
-
-    let mut final_result = Vec::<u8>::new();
-    let mut read_buffer = RefReadBuffer::new(data);
-    let mut buffer = [0; 4096];
-    let mut write_buffer = RefWriteBuffer::new(&mut buffer);
-
+    let cipher = Aes256Cbc::new_var(&key, &iv.to_be_bytes()).unwrap();
```

Comment:
```rust
let cipher = Aes256Cbc::new_var(&key, &iv.to_be_bytes()).map_err(SymmetricCipherError::InvalidLength)?;
```

Reply:
Or change the type of the result to `Result<Vec<u8>, Result<Vec<u8>, InvalidKeyIvLength>>` after re-exporting `block_modes::InvalidKeyIvLength`.

Reply:
Let's change the result type and remove all the rcrypto (rust-crypto renamed) dependencies